### DISCUSSION TITLE
Kill vpf thread

### DISCUSSION THREAD

**Discussion by Sayyam-Jain** [NVIDIA/VideoProcessingFramework#400] [Q&A]
Hi, I'm trying to use this for RTSP, unfortunately as you know it doesn't returns in case of camera/network break.

So, I was thinking to start another thread to keep track of incoming frames and kill cuda context, to free gpu memory and other resources in case of break, however I'm unable to do so.
pop or del both aren't working.

Can you please suggest something?

**Comment by rarzumanyan** [comment]
Hi @Sayyam-Jain 

Apologies for the late reply.
I assume you may use Producer-Consumer pattern.
E. g. use GStreamer of FFmpeg process for getting the compressed video frames from network add VPF process for actual decoding & processing. These processes can use FIFO channel for communication as it's shown [at this wiki page](https://github.com/NVIDIA/VideoProcessingFramework/wiki/Decoding-video-from-RTSP-camera)

P. S.
There are alternative approaches to killing a Python threads, [this](https://www.geeksforgeeks.org/python-different-ways-to-kill-a-thread/) is a nice article which covers the topic.
